common/lib/kernel/devices/generic_driver.cpp

```diff
@@ -0,0 +1,40 @@
+/* GenericDriver
+ *
+ * Author: Pascal Bauer <pascal.bauer@rwth-aachen.de>
+ *
+ * SPDX-FileCopyrightText: 2023-2024 Pascal Bauer <pascal.bauer@rwth-aachen.de>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <villas/kernel/devices/generic_driver.hpp>
+
+#include <villas/kernel/devices/device.hpp>
+#include <villas/kernel/devices/utils.hpp>
+
+using villas::kernel::devices::Device, villas::kernel::devices::GenericDriver;
+using villas::kernel::devices::utils::write_to_file;
+
+void GenericDriver::attach(const Device &device) const {
+  if (device.driver().has_value()) {
+    device.driver().value()->unbind(device);
+  }
+  this->override(device);
+  device.probe();
```

Comment:
I think the writes to the sys files can fail with a few different errors that contain information on what happened. Maybe we can put this information in an exception.